pkg/standalone-utils/contracts/relayer/GaugeActions.sol

```diff
@@ -30,13 +30,23 @@ abstract contract GaugeActions is IBaseRelayerLibrary {
     using SafeERC20 for IERC20;
 
     IBalancerMinter private immutable _balancerMinter;
+    bool private immutable _isL2Relayer;
```

Comment:
We need to know this because L1 is permissioned (can't call checkpoint directly), while L2 is not (so you can). Could this ever change in future gauge releases, or even networks? (e.g., what will happen on zkSync? Could we ever have a permissioned L2?)

Consider renaming this semantically: maybe isCheckpointPermissioned, or something like that.

Reply:
How about https://github.com/balancer/balancer-v2-monorepo/pull/2510/commits/738dc332992c5e61f4dc0918ccf7caa1160f4cc1?

L2s are not permissioned because of the version of the contract that we are using (i.e. the 'new' `ChildChainGauge`). The original contract is from Curve, but we've decided to remove that restriction -since there was always a way around it via transfers, so it was just better to just be honest about it and allow anyone to checkpoint-.

In other words, I don't think that will ever change. But I still agree that it might be better to just name the boolean according to the action.